scripts/dts/edtsdatabase.py

```diff
@@ -20,10 +20,123 @@
 class EDTSConsumerMixin(object):
     __slots__ = []
 
+    def _generate_device_dict(self, property, values):
+        devices = dict()
+        if not isinstance(values, list):
+            values = [values, ]
+        for val in values:
+            for device_id in self._edts[property].get(val, []):
+                devices[device_id] = self._edts['devices'][device_id]
+        return devices
 
-    ## Empty for now
+    ##
+    # @brief Get compatibles
+    #
+    # @param None
+    # @return edts 'compatibles' dict
+    def get_compatibles(self):
+        return self._edts['compatibles']
+
+    ##
+    # @brief Get aliases
+    #
+    # @param None
+    # @return edts 'aliases' dict
+    def get_aliases(self):
+        return self._edts['aliases']
+
+    ##
+    # @brief Get chosen
+    #
+    # @param None
+    # @return edts 'chosen' dict
+    def get_chosen(self):
+        return self._edts['chosen']
+
+    ##
+    # @brief Get device types
+    #
+    # @param None
+    # @return edts device types dict
+    def get_device_types(self):
+        return self._edts['device-types']
+
+    ##
+    # @brief Get controllers
+    #
+    # @param  None
+    # @return compatible generic device type
+    def get_controllers(self):
+        return self._edts['controllers']
+
+    ##
+    #
+
+    ##
+    # @brief Get list of device ids of a given property value
+    #
+    # @param property property
+    # @param property value(s)
+    # @return list of device ids matching given property value(s)
+    def get_device_ids_by_prop_value(self, property, value):
```

Comment:
The function description is really misleading. I tried 
```
get_device_ids_by_prop_value('label', 'SPI_1')
```
and got an exception. 
The property is restricted to only a few; this should be described.